### PR TITLE
envoy 1.16.0 (new formula)

### PR DESCRIPTION
**Pull request by ankane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

High-performance proxy

### CODE REVIEW

File: Formula/envoy.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  def install
+    args = %w[
+      --curses=no
+      --show_task_finish
+      --verbose_failures
+      --action_env=PATH=/usr/local/bin:/opt/local/bin:/usr/bin:/bin
+      --test_output=all
+    ]
+    system "bazelisk", "build", *args, "//source/exe:envoy-static"
```
Comment (SMillerDev) [inline Formula/envoy.rb:25]: Please use `Formula["bazelisk"].opt_bin"/bazelisk"` here to ensure we point to the correct formula.

**Comment by ankane** [comment]
Build failure due to AppleClang 12 (I thought the latest version addressed this, but looks like I was testing with AppleClang 11). The project is actively working on it: https://github.com/envoyproxy/envoy/pull/13364

**Comment by timsutton** [comment]
@ankane Would it make sense to also include the envoy build config option `--config=sizeopt`? In my testing that brought down the size of the static binary from ~400MB to ~80MB. Not sure if there might also be any other build options that would make sense for standard production usage.

**Comment by ankane** [comment]
@timsutton Do you know if/how that affects portability?

**Comment by timsutton** [comment]
> @timsutton Do you know if/how that affects portability?

Sadly I do not, besides it looking like this maps to c compiler optimizations. It didn't seem to change at all which frameworks were linked in, though I didn't check to see if any features changed.

**Comment by ankane** [comment]
@SMillerDev Updated to the newly released 1.16.0, which fixes the tests.

@timsutton I think it's better to go with the default build options (unless someone from the project can confirm the size optimizations don't affect portability).

**Comment by SMillerDev** [comment]
> I think it's better to go with the default build options (unless someone from the project can confirm the size optimizations don't affect portability).

We generally prefer to only use default options unless there's a strong case for adding more. I'd say 300mb storage space isn't that strong a case in a time of 1TB micro-SD cards.

**Comment by timsutton** [comment]
👍 makes sense to me to keep the default options unless there's a very strong argument to run the size optimization. @ankane Thanks for writing this formula!

**Review by SMillerDev — APPROVED** [review]
Thanks @ankane  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by ankane** [comment]
Thanks!

**Pull request closed without merging**